app/src/lib/git/stash.ts

```diff
@@ -144,23 +144,53 @@ function extractBranchFromMessage(message: string): string | null {
   return match === null || match[1].length === 0 ? null : match[1]
 }
 
-/** Get the files that were changed in the given stash commit.
- *  This is different than `getChangedFiles` because stashes
- *  have _3 parents(!!!)_
+/** The SHA for the null tree. */
+const NullTreeSHA = '4b825dc642cb6eb9a060e54bf8d69288fbee4904'
+
+/**
+ * Get the files that were changed in the given stash commit.
  *
- * TODO: support showing untracked files in the diff
+ * Runs `git diff` twice, once for tracked changes in the stash
+ * and once for untracked changes in the stash.
+ * This is different than `getChangedFiles` because stashes
+ * have _3 parents(!!!)_
  */
 export async function getStashedFiles(
   repository: Repository,
-  sha: string
+  committish: string
 ): Promise<ReadonlyArray<CommittedFileChange>> {
   // opt-in for rename detection (-M) and copies detection (-C)
   // this is equivalent to the user configuring 'diff.renames' to 'copies'
   // NOTE: order here matters - doing -M before -C means copies aren't detected
+  const baseArgs = ['diff', '-C', '-M', '--name-status', '-z']
+  const trackedArgs = [...baseArgs, committish, `${committish}^`, '--']
+  const trackedResult = await git(
+    trackedArgs,
+    repository.path,
+    'getStashedFiles (tracked)'
+  )
+
+  const trackedFiles = parseChangedFiles(trackedResult.stdout, committish)
+
+  const untrackedArgs = [...baseArgs, NullTreeSHA, `${committish}^3`, '--']
+  const untrackedResult = await git(
+    untrackedArgs,
+    repository.path,
+    'getStashedFiles (untracked)',
+    {
+      // if this fails, its most likely
+      // because there weren't any untracked files,
+      // and that's okay!
+      successExitCodes: new Set([0, 128]),
+    }
+  )
 
-  // git diff -C -M --name-status -z [STASH_SHA] [STASH_SHA]^
-  const args = ['diff', '-C', '-M', '--name-status', '-z', sha, `${sha}^`, '--']
-  const result = await git(args, repository.path, 'getStashedFiles')
+  const untrackedFiles = parseChangedFiles(
```

Comment:
I think we should only run this if `untrackedResult.exitCode === 0`, given we handle/suppress when `128` is found.

I also spotted that `untrackedResult.stdout` is an empty string here when there are no uncommitted files in the stash, so that could be another case for skipping this work...

Reply:
ooohh that's a great call. will update!

Reply:
done in b870f5582e4387feb4b4cf7282bd1d354f330545